src/AlertsAggregator.py

```diff
@@ -0,0 +1,222 @@
+            if response.json()['totalResultsCount'] == 0:
+                print(response.text)
+                return False
+            else:
+                print(response.text)
+                return True
+        else:
+            raise requests.exceptions.RequestException("An error occure while try to use get request from GeoNames api")
+
+
+    # This function display the distribution of alarms in a specific settlement
+    def display_dist(self, settlement: str):
+        try:
+            settlement_lst= self.create_user_settl_lst(settlement)
+            is_settlement_exist = self.does_the_settlement_exist(settlement)
+            if settlement_lst != []:
+                # Filtered the df by the given settlement lst
+                filtered_df = self.df[self.df['data'].isin(settlement_lst)]
+                
+                # Access the 'time' column of the DataFrame
+                filtered_df['time'] = pd.to_datetime(filtered_df['time'])
+                filtered_df['hours'] = filtered_df['time'].dt.strftime("%H")
+                hour_lst = ["00","01","02","03","04","05","06","07","08","09","10","11","12","13","14","15","16","17","18","19","20","21","22","23"]
+                alert_distribution = filtered_df['hours'].value_counts().reindex(hour_lst, fill_value=0)
+                alert_distribution = alert_distribution.to_dict()
+                distribution_lst = []
+                for i in hour_lst:
+                    alert_count_obj = AlertCountPerDay(hour=i, count=alert_distribution[i])
+                    distribution_lst.append(alert_count_obj)
+                return distribution_lst
+            elif is_settlement_exist:
+                return {"message" : f"There were no alarms in the settlement: {settlement}"}   
+            else:
+                raise ValueError("You selected a Settlement that does not exist.")
+        
+        except requests.exceptions.RequestException as req_err:
+            raise requests.exceptions.RequestException(f"Error occurred: {req_err}")
+        
+        except WrongSettlementException as settl_exc:
+            raise WrongSettlementException(settl_exc)
+        
+        except ValueError as val_err:
+            raise ValueError(val_err)   
+        
+        
+
+    # This function receives df and settlement name then count the total amount of alerts in this settlement
+    def alerts_count(self, settlement: str):
+        try:
+            user_settlement_lst= self.create_user_settl_lst(settlement)
+
+            is_settlement_exist = self.does_the_settlement_exist(settlement)
+            print(is_settlement_exist)
+            if user_settlement_lst != []:
+                # Filtered the df by the given settlement lst
+                filtered_df = self.df[self.df['data'].isin(user_settlement_lst)]
+                alert_amount = filtered_df.shape[0]
+                return {"message" : f"The total amount of alerts in {settlement} is: {alert_amount}"}
```

Comment:
this function is badly desinged.
why whould it return json message??

AlertsAggregator is not aware of the fact that there is an API!
this function sould return **int** ! with the alerts count.
if you want to create some message for some user, it shouldnt be here